shaky/src/main/java/com/linkedin/android/shaky/ShakyFlowCallback.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.android.shaky;
+
+import android.app.Activity;
+
+import androidx.annotation.IntDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Callbacks during the feedback submission flow.
+ */
+public interface ShakyFlowCallback {
+    @IntDef({SHAKY_STARTED_BY_SHAKE, SHAKY_STARTED_MANUALLY})
+    @Retention(RetentionPolicy.SOURCE)
+    @interface ShakyStartedReason {
+    }
+
+    /**
+     * The flow is started, because the device is shaken.
+     */
+    int SHAKY_STARTED_BY_SHAKE = 1;
+    /**
+     * The flow is started, because {@link Shaky#startFeedbackFlow()} is called.
+     */
+    int SHAKY_STARTED_MANUALLY = 2;
+
+    @IntDef({
+            SHAKY_FINISHED_TOO_FREQUENT,
+            SHAKY_FINISHED_NO_RESUMED_ACTIVITY,
+            SHAKY_FINISHED_ALREADY_STARTED,
+            SHAKY_FINISHED_BY_USER,
+            SHAKY_FINISHED_SENSITIVITY_UPDATED,
+            SHAKY_FINISHED_SUBMITTED
+    })
+    @Retention(RetentionPolicy.SOURCE)
+    @interface ShakeIgnoredReason {
```

Comment:
This name is a little misleading since it also includes reasons where shaky wasn't ignored, right? I'd expect it to be `ShakyFinishedReason`